x-pack/legacy/plugins/code/public/components/integrations/code_flyout_main.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, ReactNode } from 'react';
+import {
+  EuiTitle,
+  EuiFlyoutHeader,
+  EuiText,
+  EuiTabs,
+  EuiTab,
+  EuiIcon,
+  EuiLink,
+} from '@elastic/eui';
+import { CodeViewer } from './code_viewer';
+import { History } from './history';
+
+enum Tab {
+  code,
+  history,
+  blame,
+}
+
+export const CodeFlyoutMain = (props: { repo: string; file: string; revision: string }) => {
+  const [selectedTab, setSelectedTab] = React.useState<Tab>(Tab.code);
+
+  let content: ReactNode;
+
+  switch (selectedTab) {
+    case Tab.blame:
+      content = (
+        <CodeViewer
+          key="blame"
+          repo={props.repo}
+          file={props.file}
+          revision={props.revision}
+          showBlame={true}
+        />
+      );
+      break;
+    case Tab.history:
+      content = <History revision={props.revision} repo={props.repo} file={props.file} />;
+      break;
+    case Tab.code:
+      content = (
+        <CodeViewer key="code" repo={props.repo} file={props.file} revision={props.revision} />
+      );
+      break;
+  }
+
+  return (
+    <Fragment>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2 id="flyoutTitle">File Preview</h2>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <div className="codeFlyout__header">
+        <div className="codeFlyout__header--file">
+          <EuiIcon type="codeApp" size="m" />
+          <EuiText size="s">
+            <EuiLink href={`/app/code#/${props.repo}/blob/${props.revision}/${props.file}`}>
```

Comment:
@spacedragon I don't think this is generating the correct route, at least not in dev mode (the three-letter path prefix is lost)

Reply:
Hmm, it works on my machine. What url did it give you there?  
We can fire an issue after merge this PR if there is a problem.